differ/mutationDiffer.go

```diff
@@ -816,18 +823,28 @@ func NewBatch(dw *DifferWorker, startIndex, endIndex int) *batch {
 		if _, exists := b.sourceResults[fetchItem.SrcColId]; !exists {
 			b.sourceResults[fetchItem.SrcColId] = make(map[string]Result)
 		}
-		if dw.differ.compareBody {
+		switch dw.differ.compareType {
+		case base.MutationCompareTypeBodyOnly:
+			fallthrough
+		case base.MutationCompareTypeBodyAndMeta:
 			b.sourceResults[fetchItem.SrcColId][fetchItem.Key] = &GetResult{}
-		} else {
+		case base.MutationCompareTypeMetadata:
+			b.sourceResults[fetchItem.SrcColId][fetchItem.Key] = &GetMetaResult{}
+		default:
 			b.sourceResults[fetchItem.SrcColId][fetchItem.Key] = &GetMetaResult{}
 		}
 		for _, tgtColId := range fetchItem.TgtColIds {
 			if _, exists := b.targetResults[tgtColId]; !exists {
 				b.targetResults[tgtColId] = make(map[string]Result)
 			}
-			if dw.differ.compareBody {
+			switch dw.differ.compareType {
+			case base.MutationCompareTypeBodyOnly:
+				fallthrough
+			case base.MutationCompareTypeBodyAndMeta:
 				b.targetResults[tgtColId][fetchItem.Key] = &GetResult{}
-			} else {
+			case base.MutationCompareTypeMetadata:
+				b.targetResults[tgtColId][fetchItem.Key] = &GetMetaResult{}
+			default:
```

Comment:
We should probably fail instead of assuming `GetMetaResult` is usable
It'll be up to previous code to do input check before it gets to this point

Reply:
Changed to panic since input check is done in main() and this will not happen.